build_tools/kokoro/gcp_ubuntu/cmake/linux/riscv64/test.sh

```diff
@@ -18,41 +18,58 @@ export PS4='[$(date -u "+%T %Z")] '
-pushd "${BUILD_RISCV_DIR?}/iree/samples/simple_embedding" > /dev/null
 
-"${QEMU_RV64_BIN?}" -cpu rv64,x-v=true,x-k=true,vlen=256,elen=64,vext_spec=v1.0 \
--L "${RISCV_TOOLCHAIN_ROOT?}/sysroot" simple_embedding_dylib
+function generate_dylib_vmfb {
+  local target="${1}"; shift
+  local translate_arg=(
+    -iree-mlir-to-vm-bytecode-module -iree-hal-target-backends=dylib-llvm-aot
+    -iree-llvm-embedded-linker-path=${BUILD_HOST_DIR?}/install/bin/lld
+    -iree-llvm-target-triple=riscv64
+    -iree-llvm-target-cpu=generic-rv64
+    -iree-llvm-target-abi=lp64d
+  )
+  if [[ "${target}" == "mhlo" ]]; then
+    translate_arg+=(
+      -iree-input-type=mhlo
+      -iree-llvm-target-cpu-features="+m,+a,+f,+d,+c"
+    )
+  elif [[ "${target}" == "tosa" ]]; then
+    local input_file="${1}"; shift
+    iree-import-tflite -o "${BUILD_RISCV_DIR?}/tosa.mlir" "${input_file}"
+    translate_arg+=(
+      -iree-input-type=tosa
+      -iree-llvm-target-cpu-features="+m,+a,+f,+d,+c"
+      "${BUILD_RISCV_DIR?}/tosa.mlir"
+    )
+  elif [[ "${target}" == "tosa-rvv" ]]; then
+    local input_file="${1}"; shift
+    iree-import-tflite -o "${BUILD_RISCV_DIR?}/tosa.mlir" "${input_file}"
+    translate_arg+=(
+      -iree-input-type=tosa
+      -iree-llvm-target-cpu-features="+m,+a,+f,+d,+c,+v"
+      -riscv-v-fixed-length-vector-lmul-max=8
+      -riscv-v-vector-bits-min=256
+      -riscv-v-fixed-length-vector-elen-max=64
+      "${BUILD_RISCV_DIR?}/tosa.mlir"
+    )
+  fi
+  "${BUILD_HOST_DIR?}/install/bin/iree-translate" "${translate_arg[@]}" "$@"
+}
 
-"${QEMU_RV64_BIN?}" -cpu rv64,x-v=true,x-k=true,vlen=256,elen=64,vext_spec=v1.0 \
--L "${RISCV_TOOLCHAIN_ROOT?}/sysroot" simple_embedding_embedded_sync
-
-"${QEMU_RV64_BIN?}" -cpu rv64,x-v=true,x-k=true,vlen=256,elen=64,vext_spec=v1.0 \
--L "${RISCV_TOOLCHAIN_ROOT?}/sysroot" simple_embedding_vmvx_sync
-
-popd > /dev/null
-
-echo "Test e2e mlir --> bytecode module --> iree-run-module"
-
-"${BUILD_HOST_DIR?}/install/bin/iree-translate" \
-  -iree-input-type=mhlo \
-  -iree-mlir-to-vm-bytecode-module -iree-hal-target-backends=dylib-llvm-aot \
-  -iree-llvm-embedded-linker-path=${BUILD_HOST_DIR?}/install/bin/lld \
-  -iree-llvm-target-triple=riscv64 \
-  -iree-llvm-target-cpu=generic-rv64 \
-  -iree-llvm-target-cpu-features="+m,+a,+f,+d,+c" \
-  -iree-llvm-target-abi=lp64d \
+generate_dylib_vmfb mhlo \
   "${ROOT_DIR?}/iree/tools/test/iree-run-module.mlir" \
   -o "${BUILD_RISCV_DIR?}/iree-run-module-llvm_aot.vmfb"
 
-IREE_RUN_OUT=$(${QEMU_RV64_BIN?} -cpu rv64,x-v=true,x-k=true,vlen=256,elen=64,vext_spec=v1.0 \
-    -L "${RISCV_TOOLCHAIN_ROOT?}/sysroot" \
-    "${BUILD_RISCV_DIR?}/iree/tools/iree-run-module" --driver=dylib \
-    --module_file="${BUILD_RISCV_DIR?}/iree-run-module-llvm_aot.vmfb" \
-    --entry_function=abs --function_input="f32=-10")
+wget -P "${BUILD_RISCV_DIR?}/" https://github.com/tensorflow/tflite-micro/raw/aeac6f39e5c7475cea20c54e86d41e3a38312546/tensorflow/lite/micro/models/person_detect.tflite
```

Comment:
:-( I still don't like the network access here, but I know we're already doing it in other builds

Reply:
Understood. We can potentially move this flatbuffer to the docker image, but it would make the test less flexible.

Reply:
Nah baking it into the image isn't what I mean. More like decoupling artifact fetching from testing so that tests aren't flakey. Our overall build is always going to involve some network communication (obviously everything has to get to the builder in the first place), but we can put it at a different layer where we have a single place to manage retries and such. We should also be able to have it cached and replicated to further reduce the likelihood of flakes

Reply:
(but all that is a lot to cram into every test)